fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -1063,9 +1063,10 @@ public CommandProcessingResult adjustLoanTransaction(final Long loanId, final Lo
                     transactionId);
         }
 
-        if (transactionToAdjust.isChargeback()) {
-            throw new PlatformServiceUnavailableException("error.msg.loan.transaction.update.not.allowed", "Loan transaction:"
-                    + transactionId + " update not allowed as loan transaction is charge back and is linked to other transaction",
+        if (transactionToAdjust.isChargeback() || transactionToAdjust.isGoodwillCredit() || transactionToAdjust.isPayoutRefund()
```

Comment:
This is not correct. We should let the undo, but the adjusting.
Please amend this logic whether it is just undo (transactionAmount=0) or undo and create new (transactionAmount>0)